src/Api/Orbit.cpp

```diff
@@ -75,4 +76,47 @@ void orbit_api_async_string(const char* str, uint64_t id, orbit_api_color color)
     str += chunk_size;
   }
 }
+
+static void orbit_api_initialize_v0(uint64_t address) {
+  // The api function table is accessed by user code using this pattern:
+  //
+  // if(api->active && api->orbit_api_function_name) api->orbit_api_function_name(...);
+  //
+  // We don't use locking nor memory barriers here for these reasons:
+  // 1. We rely on the fact that 64 bit read/writes on aligned uint64_t are atomic on x64 so we
+  //    assume there cannot be torn reads/writes on a function address in the table.
+  // 2. Nothing bad happens if "active" is 0 but other threads see non-null functions pointers.
+  // 3. Nothing bad happens if "active" is 1 but other threads see a null function pointer.
+  //
+  // If these assumptions became false, then we should revisit this mechanism both in this function
+  // and in the user code.
+
+  orbit_api_v0* api = absl::bit_cast<orbit_api_v0*>(address);
+  if (api->initialized != 0) return;
+  api->start = &orbit_api_start;
+  api->stop = &orbit_api_stop;
+  api->start_async = &orbit_api_start_async;
+  api->stop_async = &orbit_api_stop_async;
+  api->async_string = &orbit_api_async_string;
+  api->track_int = &orbit_api_track_int;
+  api->track_int64 = &orbit_api_track_int64;
+  api->track_uint = &orbit_api_track_uint;
+  api->track_uint64 = &orbit_api_track_uint64;
+  api->track_float = &orbit_api_track_float;
+  api->track_double = &orbit_api_track_double;
+  api->initialized = 1;
+  api->active = 1;
+}
+
+void orbit_api_initialize(uint64_t address, uint64_t api_version) {
```

Comment:
Not sure if I understand this correctly: 
`address` is a `orbit_api_v0*`, right? Why pass it around as a uint64_t?

Reply:
I should add more comments. "orbit_api_initialized" is (will be) called from OrbitService with the address of the function tables to patch. For a given process, there could be different tables with different api versions if different shared libraries use the api.